src/components/form/preview.vue

```diff
@@ -0,0 +1,112 @@
+<!--
+Copyright 2024 ODK Central Developers
+See the NOTICE file at the top-level directory of this distribution and at
+https://github.com/getodk/central-frontend/blob/master/NOTICE.
+
+This file is part of ODK Central. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+
+<template>
+      <template v-if="formVersionXml.dataExists">
+        <OdkWebForm :form-xml="formVersionXml.data" @submit="handleSubmit"/>
+      </template>
+
+      <modal v-bind="previewModal" hideable backdrop @hide="previewModal.hide()">
+          <template #title>{{ $t('webFormPreview.submissionModal.title') }}</template>
+          <template #body>
+            {{ $t('webFormPreview.submissionModal.body') }}
+            <div class="modal-actions">
+              <button type="button" class="btn btn-primary" @click="closeModal()">
+                {{ $t('action.close') }}
+              </button>
+            </div>
+          </template>
+      </modal>
+</template>
+
+<script setup>
+import { createApp, getCurrentInstance } from 'vue';
+/* eslint-disable-next-line import/no-unresolved -- not sure why eslint is complaining about it */
+import { OdkWebForm, webFormsPlugin } from '@getodk/web-forms';
+import useForm from '../../request-data/form';
+import { apiPaths } from '../../util/request';
+import { noop } from '../../util/util';
+import Modal from '../modal.vue';
+import { modalData } from '../../util/reactivity';
+
+// Install WebFormsPlugin in the component instead of installing it at the
+// application level so that @getodk/web-forms package is not loaded for every
+// page, thus increasing the initial bundle
+const app = createApp({});
+app.use(webFormsPlugin);
+const inst = getCurrentInstance();
+// webFormsPlugin just adds config property to the appContext
+inst.appContext.config = app._context.config;
+
+defineOptions({
+  name: 'FormPreview'
+});
+
+const props = defineProps({
+  projectId: {
+    type: String,
+    required: true
+  },
+  xmlFormId: {
+    type: String,
+    required: true
+  },
+  draft: {
+    type: Boolean,
+    required: true
+  }
+});
+
+const { form, formVersionXml } = useForm();
+
+const previewModal = modalData();
+
+const fetchForm = () => {
+  form.request({ url: apiPaths.form(props.projectId, props.xmlFormId), extended: true })
+    .then(() => formVersionXml.request({ url: apiPaths.formXml(props.projectId, props.xmlFormId, props.draft) }))
+    .catch(noop);
```

Comment:
I have added `alert: false` for Form request so that we don't show error alert for it and made the use of Form name in the title conditional in routes.js